src/functions/questionnaire/update-questionnaire/handler.ts

```diff
@@ -0,0 +1,50 @@
+import type { ValidatedEventAPIGatewayProxyEvent } from "@libs/api-gateway";
+import { middyfy } from "@libs/lambda";
+import { questionnaireService } from "src/database/services";
+import type QuestionnaireModel from "src/database/models/questionnaire";
+import type questionnaireSchema from "src/schema/questionnaire";
+
+/**
+ * Lambda function to update a questionnaire
+ *
+ * @param event event
+ */
+const updateQuestionnaire: ValidatedEventAPIGatewayProxyEvent<typeof questionnaireSchema> = async event => {
+  const { pathParameters, body, } = event;
+  const id = pathParameters?.id;
+
+  if (!id) {
+    return {
+      statusCode: 400,
+      body: "Bad request"
+    };
+  }
+
+  const existingQuestionnaire = await questionnaireService.findQuestionnaire(id);
+  if (!existingQuestionnaire) {
+    return {
+      statusCode: 404,
+      body: `Questionnaire ${id} not found`
+    };
+  }
+
+  const questionnaireUpdates: QuestionnaireModel = {
+    ...existingQuestionnaire,
+    ...body
+  };
```

Comment:
We should allocate to each attribute specifically here. The UI will fetch the existing questionnaire to be updated, the user will update the given fields, and then the whole questionnaire will be sent to the lambda as is. In this case probably only id needs to come from the existingQuestionnaire (although you have that from the lambda event as id also, i'd still use if from existingQuestionnaire).

Here is an example of what I mean from updateCurations:
  const curationUpdates: CurationModel = {
    id: existingCuration.id,
    name: name,
    elasticCurationId: existingCuration.elasticCurationId,
    documentId: existingCuration.documentId,
    queries: queries,
    promoted: curationType === CurationType.CUSTOM ? existingCuration.promoted : promoted,
    hidden: curationType === CurationType.CUSTOM ? existingCuration.hidden : hidden,
    startTime: startTime,
    endTime: endTime,
    curationType: curationType,
    groupId: groupId,
    language: language,
    invalidDocuments: invalidDocuments
  };
  
  In this case the attributes e.g. name, queries etc have already been destructured from the body, you should do similar